lib/tests/test_eol.rs

```diff
@@ -373,6 +373,176 @@ fn test_eol_conversion_none_snapshot_conflicts() {
+    parent2_contents: "b\n",
+    extra_setting: r#"working-copy.eol-conversion = "none""#,
+    expected_eol: "\n",
+    expected_conflict_side1: "a\n",
+    expected_conflict_side2: "b\n",
+}; "LF parents with none settings")]
+#[test_case(UpdateConflictsTestConfig {
+    parent1_contents: "a\n",
+    parent2_contents: "b\n",
+    extra_setting: r#"working-copy.eol-conversion = "input-output""#,
+    expected_eol: "\r\n",
+    expected_conflict_side1: "a\r\n",
+    expected_conflict_side2: "b\r\n",
+}; "LF parents with input-output settings")]
+#[test_case(UpdateConflictsTestConfig {
+    parent1_contents: "a\r\n",
+    parent2_contents: "b\r\n",
+    extra_setting: r#"working-copy.eol-conversion = "input-output""#,
+    expected_eol: "\r\n",
+    expected_conflict_side1: "a\r\n",
+    expected_conflict_side2: "b\r\n",
+}; "CRLF parents with input-output settings")]
+fn test_eol_conversion_update_conflicts(
+    UpdateConflictsTestConfig {
+        parent1_contents,
+        parent2_contents,
+        extra_setting,
+        expected_eol,
+        expected_conflict_side1,
+        expected_conflict_side2,
+    }: UpdateConflictsTestConfig,
+) {
+    // Create a conflict commit with 2 given contents on one file, checkout that
+    // conflict with the given EOL conversion settings, and test if the EOL matches.
+
+    // Use the working-copy.eol-conversion = "none" setting to write files to the
+    // store as is.
+    let no_eol_conversion_settings =
+        base_user_settings_with_extra_configs("working-copy.eol-conversion = \"none\"\n");
+    let mut test_workspace = TestWorkspace::init_with_backend_and_settings(
+        TestRepoBackend::Git,
+        &no_eol_conversion_settings,
+    );
+    let file_repo_path = repo_path("test-eol-file");
+    let file_disk_path = file_repo_path
+        .to_fs_path(test_workspace.workspace.workspace_root())
+        .unwrap();
+
+    // The commit graph:
+    // C (conflict)
+    // |\
+    // A B
+    // |/
+    // (empty)
+    let root_commit = test_workspace.repo.store().root_commit();
+    testutils::write_working_copy_file(
+        test_workspace.workspace.workspace_root(),
+        file_repo_path,
+        parent1_contents,
+    );
+    let tree = test_workspace.snapshot().unwrap();
+    let mut tx = test_workspace.repo.start_transaction();
+    let parent1_commit = tx
+        .repo_mut()
+        .new_commit(vec![root_commit.id().clone()], tree.id())
+        .write()
+        .unwrap();
+    tx.commit("commit parent 1").unwrap();
+
+    test_workspace
+        .workspace
+        .check_out(
+            test_workspace.repo.op_id().clone(),
+            None,
+            &root_commit,
+            &CheckoutOptions::empty_for_test(),
+        )
+        .unwrap();
+    testutils::write_working_copy_file(
+        test_workspace.workspace.workspace_root(),
+        file_repo_path,
+        parent2_contents,
+    );
+    let tree = test_workspace.snapshot().unwrap();
```

Comment:
These `check_out()` and `snapshot()` look redundant. Maybe you can just write trees to repo, and check out the merge commit at the end.

Reply:
Get rid of intermmediate `checkout` and `snapshot`, and directly creates the tree in the repo.